src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BazelModTidyFunction.java

```diff
@@ -94,41 +93,30 @@ public SkyValue compute(SkyKey skyKey, Environment env)
             .getOrDefault(ModuleKey.ROOT, ImmutableMap.of())
             .keySet()
             .stream()
-            .map(SingleExtensionEvalValue::key)
+            // Use the unvalidated key to avoid errors caused by incorrect imports - we can fix
+            // them.
+            .map(SingleExtensionValue::evalKey)
             .collect(toImmutableSet());
     SkyframeLookupResult result = env.getValuesAndExceptions(extensionsUsedByRootModule);
     if (env.valuesMissing()) {
       return null;
     }
+    ImmutableList.Builder<RootModuleFileFixup> fixups = ImmutableList.builder();
     for (SkyKey extension : extensionsUsedByRootModule) {
-      try {
-        result.getOrThrow(extension, ExternalDepsException.class);
-      } catch (ExternalDepsException e) {
-        if (e.getDetailedExitCode().getFailureDetail() == null
-            || !e.getDetailedExitCode()
-                .getFailureDetail()
-                .getExternalDeps()
-                .getCode()
-                .equals(FailureDetails.ExternalDeps.Code.INVALID_EXTENSION_IMPORT)) {
-          throw new BazelModTidyFunctionException(e, SkyFunctionException.Transience.PERSISTENT);
-        }
-        // This is an error bazel mod tidy can fix, so don't fail.
+      if (result.get(extension) instanceof SingleExtensionValue evalValue) {
```

Comment:
this is much nicer than the old try/catch :)

nit, though -- I see what you did there with the `instanceof` :) but IMO it's better to be slightly more verbose there and check for null and then do a cast.